pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -522,7 +525,32 @@ func (i *inode) NewFile(ctx context.Context, name string, opts vfs.OpenOptions)
 		},
 		Name: linux.CString(name),
 	}
-	return i.newEntry(ctx, name, linux.S_IFREG, linux.FUSE_CREATE, &in)
+
+	// Here a file descriptor is returned by fuse server at a time of file creating.
+	// So it should be returned to caller or released to avoid Fh leak.
+	// The best option would be returning file handle by NewFile(...) but
+	// that means filesystem interface should be changed.
+	// Here we just releasing received file descriptor immediately to avoid leakage.
```

Comment:
Thanks for the good idea about storing Fh in inode until it returned by Open, fixed in that way.

> To avoid ultimately leaking the Fh, the inode could clean up any cached Fhs when it reaches zero refs.
In this case Fh wasn't cached as gvisor was ignoring returned Fh, leak was on a fuse server side.